src/mbgl/renderer/render_tile.cpp

```diff
@@ -60,6 +61,37 @@ mat4 RenderTile::translatedClipMatrix(const std::array<float, 2>& translation,
     return translateVtxMatrix(nearClippedMatrix, translation, anchor, state, false);
 }
 
+const OverscaledTileID& RenderTile::getOverscaledTileID() const { return tile.id; }
+bool RenderTile::holdForFade() const { return tile.holdForFade(); }
+
+Bucket* RenderTile::getBucket(const style::Layer::Impl& impl) const {
+    return tile.getBucket(impl);
+}
+
+const LayerRenderData* RenderTile::getLayerRenderData(const style::Layer::Impl& impl) const {
+    return tile.getLayerRenderData(impl);
+}
+
+optional<ImagePosition> RenderTile::getPattern(const std::string& pattern) const {
+    assert(tile.kind == Tile::Kind::Geometry);
+    return static_cast<const GeometryTile&>(tile).getPattern(pattern);
+}
+
+const optional<gfx::Texture>& RenderTile::getGlyphAtlasTexture() const {
+    assert(tile.kind == Tile::Kind::Geometry);
+    return static_cast<const GeometryTile&>(tile).glyphAtlasTexture;
+}
+
+const optional<gfx::Texture>& RenderTile::getIconAtlasTexture() const {
```

Comment:
we apparently never really check for `nullopt`, so why is this optional?

Reply:
`GeometryTile` is keeping those as optional, so just taking after... but I agree, here we can return just `Texture`, and assert that it had been assigned. Thanks for noticing!